### ISSUE TITLE
"www." crashes with isDomainName at DomainUtils._getLeftPart

### ISSUE THREAD

**Issue by modulovalue** [Ephenodrom/Dart-Basic-Utils#6]
"www." crashes with isDomainName at DomainUtils._getLeftPart

**Comment by Ephenodrom** [comment]
Hello @modulovalue 

The bug is fixed with version 2.3.5. Already updated it on pub.dev.

I now check if the given domain name starts with "*." ( commonly used if you have a wildcard certificate ) or the given string is a subdomain ( the www. case ). if one of these two conditions matches, i will return false. 

Regards and thanks for using this package.

**Comment by modulovalue** [comment]
@Ephenodrom are you sure that it was fixed?

I get 
```
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════      
flutter: The following RangeError was thrown building HookBuilder(dirty, state: _HookWidgetState#42555):           
flutter: Value not in range: -1      
```

at
```
When the exception was thrown, this was the stack:                                                        
flutter: #0      _StringBase.substring (dart:core-patch/string_patch.dart:391:7)                                   
flutter: #1      DomainUtils._getLeftPart (package:basic_utils/src/DomainUtils.dart:240:23)                        
flutter: #2      DomainUtils.parseDomain (package:basic_utils/src/DomainUtils.dart:160:23)                         
flutter: #3      DomainUtils.isDomainName (package:basic_utils/src/DomainUtils.dart:21:12)     
```
on 2.3.5

for the input "www."

**Comment by Ephenodrom** [comment]
@modulovalue 
Ah i see, i was thinking of a domainname that starts with "www." like "www.domain.de". 
Fix it now with version 2.3.6 and released it on pub.dev.

If there are any other issues feel free to open an issue :).

Regards
